### DISCUSSION TITLE
advice for connection handling in aws lambda

### DISCUSSION THREAD

**Discussion by chifi49** [planetscale/discussion#73] [Documentation] [answered]
I have 2 questions 
a. First I need advice for the correct way to use a connection in aws lambda
   Currently we initialize the connection before exports.handler function is invoked.
   Is it safe for a connection to be used like this? (since it is initialized before the exports.handler it will be used multiple times for an unknown number of lambda invocations)

    The following code is used as an example
    
    /**aws lambda code **/
    const { PSDB } = require('planetscale-node')

     const conn = new PSDB('main')
     async function count(){
          return await conn.query("select count(1) from test");
    }
    exports.handler = async (event) => {
         const [rows, fields] = await count();
         console.log(rows);
        const response = {
           statusCode: 200,
           body: JSON.stringify(rows),
       };
       return response;
     };

2. I have noticed in the code of planetscale-node you make a request to api.planetscale "with the function postJSON" and receive the following fields
certificate, certificate_chain, remote_addr and ports
For how long are these remain valid to used in a connection to mysql.createConnection. Can we cache them for e.x. for 1 day, after how long they expire?

**Comment by mscoutermarsh** [comment, marked as answer]
Hi @chifi49,

a. Yes that is safe and the recommended way. PlanetScale can handle 10's of thousands of connections. You don't need to worry about hitting a connection limit.

b. They are valid for 7 days. Caching for 1 day would work great. Also, we're working on another method of connection which won't require the API call. Will be available soon.
